plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,368 @@
+        }
+        // check for nested array
+        else {
+            if (originalFieldName.contains(".")) {
+
+                Map<String, Object> sourceObject = ingestDocument.getSourceAndMetadata();
+                Configuration suppressExceptionConfiguration = Configuration
+                    .builder()
+                    .options(Option.SUPPRESS_EXCEPTIONS, Option.DEFAULT_PATH_LEAF_TO_NULL, Option.ALWAYS_RETURN_LIST)
+                    .build();
+                String rewriteDotPathForNestedObject = findDotPathForNestedObject(sourceObject, originalFieldName);
+                ArrayList<Object> fieldValueList = JsonPath
+                    .using(suppressExceptionConfiguration)
+                    .parse(sourceObject)
+                    .read(rewriteDotPathForNestedObject);
+
+                String originalFieldValueAsString = toString(fieldValueList);
+                updateModelParameters(ModelInputFieldName, originalFieldValueAsString, modelParameters);
+            }
+        }
+    }
+
+    /**
+     * support multiple document fields to map to the same model input fields,
+     * check if the key existed, then append to existed value to be a list.
+     */
+    private void updateModelParameters(String ModelInputFieldName, String originalFieldValueAsString, Map<String, String> modelParameters) {
+
+        if (modelParameters.containsKey(ModelInputFieldName)) {
+            Object existingValue = modelParameters.get(ModelInputFieldName);
+            List<Object> updatedList;
+
+            if (!(existingValue instanceof List)) {
+                updatedList = new ArrayList<>();
+                updatedList.add(existingValue);
+            } else {
+                updatedList = (List<Object>) existingValue;
+            }
+
+            updatedList.add(originalFieldValueAsString);
+
+            modelParameters.put(ModelInputFieldName, toString(updatedList));
+        } else {
+            modelParameters.put(ModelInputFieldName, originalFieldValueAsString);
+        }
+
+    }
+
+    private String getFieldPath(IngestDocument ingestDocument, String originalFieldName) {
+        final boolean fieldPathIsNullOrEmpty = Strings.isNullOrEmpty(originalFieldName);
+
+        if (fieldPathIsNullOrEmpty) {
+            return null;
+        }
+        final boolean hasFieldPath = ingestDocument.hasField(originalFieldName, true);
+        if (!hasFieldPath) {
+            return null;
+        }
+        return originalFieldName;
+    }
+
+    /**
+     * appendFieldValue to the ingestDocument without changing the source
+     */
+
+    private void appendFieldValue(
+        ModelTensorOutput modelTensorOutput,
+        String originalModelOutputFieldName,
+        String newModelOutputFieldName,
+        IngestDocument ingestDocument
+    ) {
+        Object modelOutputValue = null;
+
+        Map<String, ?> modelTensorOutputMap = modelTensorOutput.getMlModelOutputs().get(0).getMlModelTensors().get(0).getDataAsMap();
```

Comment:
just wanted to confirm it's guaranteed we will have an element here, otherwise, get(0) could be troublesome.

Reply:
added the check to check if  modelTensorOutput.getMlModelOutputs() is null